src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java

```diff
@@ -560,26 +518,36 @@ BlobMetadata buildAndUpload(String blobPath, List<List<ChannelData<T>>> blobData
 
     blob.blobStats.setBuildDurationMs(buildContext);
 
-    return upload(blobPath, blob.blobBytes, blob.chunksMetadataList, blob.blobStats);
+    return upload(
+        this.storageManager.getStorage(blobData.get(0).get(0).getChannelContext()),
```

Comment:
lets chat about this f2f. Looks ripe for a nullref or a logical bug where the wrong channel context gets passed in.
Ideally you want to pass the channelContext into this method as an argument and not code in this get(0).get(0) assumption.

Reply:
Moved the getting channel context logic to call site.